.github/workflows/reusable-version-info.yml

```diff
@@ -38,7 +44,13 @@ jobs:
         id: set_outputs
         shell: bash -l {0}
         run: |
-          git fetch origin +refs/tags/*:refs/tags/*
-          export SDIST_VERSION=$(python setup.py --version)
-          echo "::set-output name=version::${SDIST_VERSION}"
-          echo "::set-output name=version_tag::${SDIST_VERSION/+/_}"
+          if [ "${{ inputs.method }}" == "setup.py" ]; then
+            export SDIST_VERSION=$(python setup.py --version)
+          elif [ "${{ inputs.method }}" == "setuptools_scm" ]; then
+            export SDIST_VERSION=$(python -m setuptools_scm)
+          elif [ "${{ inputs.method }}" == "git" ]; then
+            export SDIST_VERSION=$(git describe --dirty --tags --long --match "*[0-9]*")
+          fi
+          echo "Version number: ${SDIST_VERSION}"
+          echo "version=${SDIST_VERSION}" >> $GITHUB_OUTPUT
+          echo "version_tag=${SDIST_VERSION/+/_}" >> $GITHUB_OUTPUT
```

Comment:
These last two lines changed because of a pending depreciation of `set-output`:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/